lib/semantics/mod-file.cc

```diff
@@ -34,14 +37,16 @@ using namespace parser::literals;
 static constexpr auto extension{".mod"};
 // The initial characters of a file that identify it as a .mod file.
 static constexpr auto magic{"!mod$"};
-// Construct the path to a module file.
-static std::string ModFilePath(const std::string &, const std::string &);
+
 // Helpers for creating error messages.
 static parser::Message Error(
     const SourceName &, parser::MessageFixedText, const std::string &);
 static parser::Message Error(const SourceName &, parser::MessageFixedText,
     const std::string &, const std::string &);
 
+static const SourceName *GetSubmoduleParent(const parser::Program &);
+static std::string ModFilePath(
+    const std::string &, const SourceName &, const std::string &);
```

Comment:
These are forward declarations for the benefit of the compiler rather than the reader. They can look at the definition of the functions further down the file where there is a comment and argument names.

If this were in a .h file I would put the comment there and argument names if they help document it (like in this case).